File: RestaurantPOS.Android/app/src/main/java/com/example/restaurantpos/restaurantpo/restaurantposandroid/api/AuthApi.java
package com.example.restaurantpos.restaurantpo.restaurantposandroid.api;

import com.example.restaurantpos.restaurantpo.restaurantposandroid.models.LoginRequest;
import com.example.restaurantpos.restaurantpo.restaurantposandroid.models.LoginResponse;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.POST;

public interface AuthApi {
    
    @POST("Auth/Login")
    Call<LoginResponse> login(@Body LoginRequest request);
    
    // Future endpoints can be added here
    // @POST("Auth/Register")
    // Call<UserResponse> register(@Body RegisterRequest request);
    
    // @POST("Auth/ForgotPassword")
    // Call<Void> forgotPassword(@Body ForgotPasswordRequest request);
    
    // @POST("Auth/ResetPassword")
    // Call<Void> resetPassword(@Body ResetPasswordRequest request);
}

File: RestaurantPOS.Android/app/src/main/java/com/example/restaurantpos/restaurantpo/smartorder/data/remote/api/AuthApi.kt
package com.example.restaurantpos.restaurantpo.smartorder.data.remote.api

import com.example.restaurantpos.restaurantpo.smartorder.data.remote.dto.LoginRequest
import com.example.restaurantpos.restaurantpo.smartorder.data.remote.dto.LoginResponse
import com.example.restaurantpos.restaurantpo.smartorder.data.remote.dto.UpdateFcmTokenRequest
import retrofit2.http.Body
import retrofit2.http.POST

interface AuthApi {
    @POST("api/auth/login")
    suspend fun login(@Body request: LoginRequest): LoginResponse

    @POST("api/auth/UpdateFcmToken")
    suspend fun updateFcmToken(@Body request: UpdateFcmTokenRequest)

    @retrofit2.http.GET("api/auth/Users")
    suspend fun getUsers(): List<com.example.restaurantpos.restaurantpo.smartorder.data.remote.dto.UserDto>

    @POST("api/auth/Register")
    suspend fun register(@Body request: com.example.restaurantpos.restaurantpo.smartorder.data.remote.dto.RegisterRequest): com.example.restaurantpos.restaurantpo.smartorder.data.remote.dto.UserDto

    @retrofit2.http.PUT("api/auth/Users/{id}")
    suspend fun updateUser(@retrofit2.http.Path("id") id: Int, @Body user: com.example.restaurantpos.restaurantpo.smartorder.data.remote.dto.UserDto)

    @retrofit2.http.DELETE("api/auth/Users/{id}")
    suspend fun deleteUser(@retrofit2.http.Path("id") id: Int)
}
